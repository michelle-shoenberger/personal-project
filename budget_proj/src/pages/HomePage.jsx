import axios from 'axios';
import { useContext } from 'react';
import { useNavigate } from 'react-router-dom'
import {Helmet} from 'react-helmet'
import MyForm from "../components/MyForm";
import Summary from '../components/Summary';
import { UserContext } from '../context/UserContext';

export default function HomePage(props) {
  const {cats} = useContext(UserContext);
  const navigate = useNavigate();

  const model = {
    'item_name': 'text',
    'cost': 'float',
    'category': 'select',
    'description': 'textarea'
  };
  const handleSubmit = async (data, endpoint) => {
    //console.log(data)
    let resp = await axios.post('http://127.0.0.1:8000/api'+endpoint, data)
      .catch((e) => {
        console.log("getExpense error: " + e)
      });
    if (resp){
      //console.log(resp)
      alert("Expense successfully added.  All expenses listed on Expense page.")
      navigate('/expenses')
    } else {
      //console.log('fail')
      alert("Please enter a valid expense.")
    }
  };

  

  return (
    <>
      <Helmet>
        <title>Home</title>
      </Helmet>
      <h1 className="mb-5"> Home Page </h1>
      <Summary />
      <h2 className="mt-5"> Create new expense: </h2>
      <MyForm model={model} followOn={handleSubmit} endpoint="/expenses/" choices={cats} />
    </>
  )
}