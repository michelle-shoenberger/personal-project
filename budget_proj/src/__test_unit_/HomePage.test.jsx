import {describe, it, expect} from 'vitest'
import TestRenderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import HomePage from '../pages/HomePage'

describe("HomePage", () => {
  it("check form", () => {
    const homepage = TestRenderer.create(<MemoryRouter><HomePage /></MemoryRouter>).toJSON()
    console.log(homepage[4].children[0].props)
    const form = homepage[4];
    expect(form.children.length).toBe(5)
    expect(form.children[0].props.type).toBe('text')
  })

  it("create a snapshot", () => {
    const homepage = TestRenderer.create(<MemoryRouter><HomePage /></MemoryRouter>).toJSON()
    expect(homepage).toMatchSnapshot();
  })
})