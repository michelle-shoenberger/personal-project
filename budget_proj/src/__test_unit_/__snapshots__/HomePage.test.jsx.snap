// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > create a snapshot 1`] = `
[
  <h1
    className="mb-5"
  >
     Home Page 
  </h1>,
  <h2>
     Total expenditure this month 
  </h2>,
  <div
    className="mt-2"
  >
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>,
  <h2
    className="mt-3"
  >
     Budget status 
  </h2>,
  <div>
    <canvas
      height={150}
      role="img"
      width={300}
    />
  </div>,
  <h2
    className="mt-5"
  >
     Create new expense: 
  </h2>,
  <form
    className="myform d-flex flex-column align-items-center"
    onSubmit={[Function]}
  >
    <input
      name="item_name"
      placeholder="item_name"
      type="text"
    />
    <input
      name="cost"
      placeholder="cost"
      type="float"
    />
    <input
      name="category"
      placeholder="category"
      type="select"
    />
    <textarea
      name="description"
      placeholder="description"
    />
    <button
      className="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;
